autoscale-core/src/main/java/com/hpe/caf/autoscale/core/GovernorImpl.java

```diff
@@ -178,4 +227,72 @@ private boolean shouldBeScaledDown(final int currentMemoryLoadLimit, final int s
             }
         }
     }
+
+    private static boolean isStringNullOrEmpty(final String string)
+    {
+        return string == null || string.isEmpty();
+    }
+
+    private final class AdvancedInstanceInfo
+    {
+        private final InstanceInfo instanceInfo;
+        private int desiredInstances;
+        private double percentageDifference;
+
+        private AdvancedInstanceInfo(final InstanceInfo instanceInfo)
+        {
+            this.instanceInfo = instanceInfo;
+        }
+
+        private AdvancedInstanceInfo(final InstanceInfo instanceInfo, final int desiredInstances)
+        {
+            this.instanceInfo = instanceInfo;
+            this.desiredInstances = desiredInstances;
+        }
```

Comment:
I know that Java will set `percentageDifference` to 0 but I've always felt that constructors should initialize all instance variables.